platform/linux-generic/arch/aarch64/odp_random.h

```diff
@@ -0,0 +1,74 @@
+/* Copyright (c) ARM Limited
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:     BSD-3-Clause
+ */
+
+#ifndef ODP_AARCH64_RANDOM_H_
+#define ODP_AARCH64_RANDOM_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <odp/api/spec/random.h>
+#include <odp/autoheader_internal.h>
+
+#include <stdint.h>
+
+#define RNDR_OUTPUT_LEN      8
+
+odp_random_kind_t _odp_random_max_kind_generic(void);
+int32_t _odp_random_true_data_generic(uint8_t *buf, uint32_t len);
+int32_t _odp_random_crypto_data_generic(uint8_t *buf, uint32_t len);
+
+static inline int _odp_random_max_kind(void)
+{
+	if (_ODP_OPENSSL_RAND) {
```

Comment:
Hi @MatiasElo 
Armv8.5-a RNG extensions support 2 registers, `RNDR & RNDRRS`, both of which return TRUE random numbers according to the [specification](https://developer.arm.com/documentation/ddi0595/2021-06/AArch64-Registers/RNDR--Random-Number). There is no crypto random number generator equivalent in Arm ISA as far as I know. 
As you point out in a later comment, ODP spec states that `Implementations support the returned max kind and all kinds weaker than it.` If OpenSSL is disabled in the current setup for architecture specific random implementation, the highest supported random kind will become `ODP_RANDOM_BASIC`. `ODP_RANDOM_CRYPTO` cannot be supported without OpenSSL and `ODP_RANDOM_TRU`E may be supported but the ODP spec imposes a limitation since `ODP_RANDOM_CRYPTO`. 
So, what changes should be done to support Armv8.5-a RNG? I see two options at the moment:
1. Change the ODP to remove the limitation for max kind to support all weaker kinds
2. Change the current setup for random to architecture specific random implementation and OpenSSL concurrently

Let me know what you think.

Reply:
My understanding of the spec is that it's okay to use the higher lever random implementation also for the lower level(s). It's not a problem if the returned data is "too random".

Reply:
So, according to the spec, I can still use the Armv8.5-a RNDR extension to generate a random number but the return type for `_odp_random_max_kind()` would be `ODP_RANDOM_BASIC` when using AArch64 specific implementation since we can upgrade a random data request. Is my understanding correct?
@MatiasElo, can you please check the comment by myself @JereLeppanen below? There are different approaches to resolve this and depending on how we decide to proceed with the implementation, I'll make the changes. Thanks!

Reply:
I agree with Jere. So, `RNDR` would be used for `ODP_RANDOM_CRYPTO` and `RNDRRS` for `ODP_RANDOM_TRUE`.